test/functional/api/buffer_spec.lua

```diff
@@ -392,6 +392,89 @@ describe('api/buf', function()
     end)
   end)
 
+  describe('nvim_buf_get_lines, nvim_buf_set_text', function()
+    local get_lines, set_text = curbufmeths.get_lines, curbufmeths.set_text
+    local line_count = curbufmeths.line_count
+
+    it('works', function()
+      insert([[
+      hello foo!
```

Comment:
I'm afraid that this test indeed fails with non-ASCII characters (try it with `hallå` instead of `hello`).